### PR TITLE
[Enhancement] rewrite torch.cat for TensorRT when input is dynamic

### PR DESCRIPTION
**Pull request by AllentDan**

### CODE REVIEW

File: mmdeploy/pytorch/functions/cat.py [hunk 1]
```diff
@@ -0,0 +1,24 @@
...
+
+
+@FUNCTION_REWRITER.register_rewriter(func_name='torch.cat', backend='tensorrt')
+def cat__tensorrt(tensors: Sequence[Tensor], *args, **kwargs) -> torch.Tensor:
+    """Rewrite `cat` for TensorRT backend.
+
+    cat in TensorRT does not support bool or uint8 type when input is dynamic.
+    """
+    ctx = FUNCTION_REWRITER.get_context()
+    if get_dynamic_axes(ctx.cfg) is None:
```
Comment (grimoire) [inline mmdeploy/pytorch/functions/cat.py:18]: warning: there is a potential risk. deploy config is not required for tensorrt.from_onnx.

**Review by grimoire — APPROVED** [review]
LGTM

File: mmdeploy/pytorch/functions/cat.py (hunk 1 above)
Comment (AllentDan) [reply mmdeploy/pytorch/functions/cat.py:18]: I suppose it only works for `torch2onnx` with `dynamic_axies` not None.

**Review by tpoisonooo — APPROVED** [review]

File: mmdeploy/pytorch/functions/cat.py (hunk 1 above)
Comment (lvhan028) [reply mmdeploy/pytorch/functions/cat.py:18]: @grimoire can we handle this case in the latest refactor plan?

Comment (grimoire) [reply mmdeploy/pytorch/functions/cat.py:18]: @lvhan028 Nope

**Pull request merged**